src/main/scala/junctions/addrmap.scala

```diff
@@ -121,6 +119,16 @@ class AddrMap(
     }.flatten.sortBy(_.region.start)
   }
 
+  // checks to see whether any MemRange overlaps within this AddrMap
+  flatten.combinations(2) foreach {
+    case (Seq(AddrMapEntry(an, ar), AddrMapEntry(bn, br))) =>
+      val ar_end = ar.start + ar.size
+      val br_end = br.start + br.size
+      val ar_overlaps_br = ar.start < br_end && br.start < ar_end
+      require(!ar_overlaps_br,
```

Comment:
I still think it's more understandable to just have this be

    ar.start >= br_end || br.start >=ar_end

But I guess it's fine as is.